### PR TITLE
DOC: Draw more attention to which functions in random are convenience wrappers

### PR DESCRIPTION
**Pull request by eric-wieser**
I've seen people complain in the past that `randn(M, N)` is inconsistent with `zeros((M, N))`. They're right, but they're missing the point that `randn` exists to be consistent with Matlab, not to be self-consistent with numpy.

Let's point those users more obviously to the function they're looking for.

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -1328,6 +1328,12 @@ cdef class RandomState:
 
         Random values in a given shape.
 
+        .. note::
+            This is a convenience function for users porting code from Matlab,
+            which wraps `numpy.random.random_sample`. This inner function
+            takes a tuple to specify the size of the output, which is more
+            consistent with other NumPy functions likes `zeros` and `ones`.
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: "likes" <- "like"

File: numpy/random/mtrand/mtrand.pyx [hunk 2]
```diff
@@ -1328,6 +1328,12 @@ cdef class RandomState:
 
         Random values in a given shape.
 
+        .. note::
+            This is a convenience function for users porting code from Matlab,
+            which wraps `numpy.random.random_sample`. This inner function
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Maybe "That function"

File: numpy/random/mtrand/mtrand.pyx (hunk 1 above)
Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: Maybe "which is consistent"

File: numpy/random/mtrand/mtrand.pyx [hunk 3]
```diff
@@ -1372,16 +1372,17 @@ cdef class RandomState:
...
-        If positive, int_like or int-convertible arguments are provided,
-        `randn` generates an array of shape ``(d0, d1, ..., dn)``, filled
-        with random floats sampled from a univariate "normal" (Gaussian)
-        distribution of mean 0 and variance 1 (if any of the :math:`d_i` are
-        floats, they are first converted to integers by truncation). A single
-        float randomly sampled from the distribution is returned if no
-        argument is provided.
+        .. note::
+            This is a convenience function for users porting code from Matlab,
+            which wraps `numpy.random.standard_normal`. This inner function
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: "which wraps" <- "and". "Which" seems to refer to Matlab.

Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: See above for other comments.

File: numpy/random/mtrand/mtrand.pyx (hunk 2 above)
Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: "which wraps" <- "and". "Which" seems to refer to Matlab.

File: numpy/random/mtrand/mtrand.pyx [hunk 4]
```diff
@@ -1372,16 +1372,17 @@ cdef class RandomState:
...
+            takes a tuple to specify the size of the output, which is more
+            consistent with other NumPy functions likes `zeros` and `ones`.
 
-        This is a convenience function.  If you want an interface that takes a
-        tuple as the first argument, use `numpy.random.standard_normal` instead.
+        If positive int_like arguments are provided, `randn` generates an array
+        of shape ``(d0, d1, ..., dn)``, filled
+        with random floats sampled from a univariate "normal" (Gaussian)
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Don't need quotes around "normal".

File: numpy/random/mtrand/mtrand.pyx [hunk 5]
```diff
@@ -1372,16 +1372,17 @@ cdef class RandomState:
...
+        .. note::
+            This is a convenience function for users porting code from Matlab,
+            which wraps `numpy.random.standard_normal`. This inner function
+            takes a tuple to specify the size of the output, which is more
+            consistent with other NumPy functions likes `zeros` and `ones`.
 
-        This is a convenience function.  If you want an interface that takes a
-        tuple as the first argument, use `numpy.random.standard_normal` instead.
+        If positive int_like arguments are provided, `randn` generates an array
+        of shape ``(d0, d1, ..., dn)``, filled
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Might as well justify the text a bit more.

File: numpy/random/mtrand/mtrand.pyx [hunk 6]
```diff
@@ -1536,20 +1536,43 @@ cdef class RandomState:
         Returns
         -------
         out : float or ndarray
-            Drawn samples.
+            A ``size``-shaped array of drawn floating-point samples, or a single
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: "An array of shape ``size``..."

File: numpy/random/mtrand/mtrand.pyx [hunk 7]
```diff
@@ -1536,20 +1536,43 @@ cdef class RandomState:
...
+            sample if no ``size`` was supplied
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: "if ``size`` was not specified".

File: numpy/random/mtrand/mtrand.pyx [hunk 8]
```diff
@@ -1536,20 +1536,43 @@ cdef class RandomState:
...
+
+        Notes
+        -----
+        For random samples from :math:`N(\\mu, \\sigma^2)`, use either of::
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: "one of::"?

File: numpy/random/mtrand/mtrand.pyx (hunk 4 above)
Comment (eric-wieser) [reply numpy/random/mtrand/mtrand.pyx]: Inherited from the old docstring

File: numpy/random/mtrand/mtrand.pyx (hunk 5 above)
Comment (eric-wieser) [reply numpy/random/mtrand/mtrand.pyx]: Makes the diff noisier and the two commits harder to separate, would prefer not to

**Comment by eric-wieser** [comment]
Good suggestions, clearly I'm out of practice at writing docs.

**Comment by mattip** [comment]
Changes can be viewed at [matlab for numpy](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/user/numpy-for-matlab-users.html), [rand](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.random.rand.html), [randn](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.random.randn.html)

They seem to render nicely.

**Comment by eric-wieser** [comment]
I was hoping that zeros and ones would be links, not italic.

**Comment by mattip** [comment]
Try to add `numpy.`?
```
`numpy.ones`
```

**Comment by mattip** [comment]
Once PRs #13002, #13033 and issue #13013 go in, the next step is to turn on warnings for all the bad references

**Comment by mattip** [comment]
@charris any more comments? The new material can be viewed at [matlab for numpy](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/user/numpy-for-matlab-users.html), [rand](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.random.rand.html), [randn](https://6086-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.random.randn.html), LGTM.

**Comment by eric-wieser** [comment]
@Mattip: I'd like to fix those references, but haven't had a chance. Feel free to take over to do it, else I'll try in 10 hours or so.

**Comment by eric-wieser** [comment]
References updated

**Comment by mattip** [comment]
The `numpy.ones` and `numpy.zeros` links in the note on the [`randn'](https://6122-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.random.randn.html?highlight=randn) page now work. In it goes. Thanks Erirc

**Pull request merged**